shim/src/handle_system_kvm_get_msr_index_list.c

```diff
@@ -38,6 +46,51 @@
 NODISCARD int64_t
 handle_system_kvm_get_msr_index_list(struct kvm_msr_list *const pmut_ioctl_args) NOEXCEPT
 {
-    (void)pmut_ioctl_args;
+    uint32_t mut_nmsrs;
+    int64_t mut_i;
+
+    struct mv_rdl_t *const pmut_rdl = (struct mv_rdl_t *)shared_page_for_current_pp();
+    platform_expects(NULL != pmut_rdl);
+
+    if (detect_hypervisor()) {
+        bferror("The shim is not running in a VM. Did you forget to start MicroV?");
+        return SHIM_FAILURE;
+    }
+
+    if (!pmut_ioctl_args->indices) {
```

Comment:
The use of ! is an implicit cast. Be specific as to what "indicies" is not supposed to be.